zookeeper-client/zookeeper-client-c/src/zookeeper.c

```diff
@@ -3495,6 +3495,51 @@ static int add_multi_completion(zhandle_t *zh, int xid, void_completion_t dc,
     return add_completion(zh, xid, COMPLETION_MULTI, dc, data, 0,0, clist);
 }
 
+/**
+ * After sending the close request, we are waiting for a given millisecs for
+ * getting the answer and/or for the socket to be closed by the server.
+ *
+ * This function should not be called while we still want to process
+ * any response from the server. It must be called after adaptor_finish called,
+ * in order not to mess with the I/O receiver thread in multi-threaded mode.
+ */
+int wait_for_session_to_be_closed(zhandle_t *zh, int timeout_ms)
+{
+    int ret = 0;
+#ifndef WIN32
+    struct pollfd fd_s[1];
+#else
+    fd_set rfds;
+    struct timeval waittime = {timeout_ms / 1000, (timeout_ms % 1000) * 1000000};
```

Comment:
@symat ,
should it be ?:
struct timeval waittime = {timeout_ms / 1000, (timeout_ms % 1000) * 1000};
instead of
struct timeval waittime = {timeout_ms / 1000, (timeout_ms % 1000) * 1000000};

Reply:
You are right, nice catch, thanks! :)
timeval indeed expects microsec, not nanosec.
I will fix in the next commit.